crux-http-server/src/crux/http_server.clj

```diff
@@ -393,10 +412,40 @@
      (log/info "HTTP server started on port: " server-port)
      (->HTTPServer server options))))
 
+(defn- edn->html [edn]
+  (format
+   "<dl>%s</dl>"
+   (apply str
+          (map
+           (fn [[k v]]
+             (format "<dt>%s</dt><dd>%s</dd>" k v))
+           edn))))
+
+(defn- html-encoder [_]
+  (reify
+    mfc/EncodeToBytes
+    (encode-to-bytes [_ data charset]
+      (.getBytes
+       ^String (edn->html data)
+       ^String charset))
+    mfc/EncodeToOutputStream
+    (encode-to-output-stream [_ data charset]
```

Comment:
the reason for a separate `encode-to-output-stream` (in addition to `encode-to-bytes`) is likely that the former's to be used for large streaming response bodies (which we will likely need for queries), the latter being sufficient for when the response fits in memory.

probably more of an issue when you implement `/_query`, so maybe this is fine for now

Reply:
yep, let's leave this for query